BUG: AssertionError in test__dual_annealing.py in test_bounds_class

### Describe your issue.

While testing scipy/optimize/tests, I noticed that test_bounds_class in test__dual_annealing.py throws an AssertionError at line 365
`assert_allclose(ret_bounds_class.x, np.arange(-2, 3), atol=1e-8)`.
However, easing the absolute tolerance to say, 1e-7 lets the test pass, although I'm not sure if this is the right thing to do.



### Reproducing Code Example

```python
python runtests.py scipy/optimize/tests
```


### Error message

```shell
scipy/optimize/tests/test__dual_annealing.py:365: in test_bounds_class
    assert_allclose(ret_bounds_class.x, np.arange(-2, 3), atol=1e-8)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=1e-08
E   
E   Mismatched elements: 1 / 5 (20%)
E   Max absolute difference: 1.0046973e-08
E   Max relative difference: 0.
E    x: array([-2.000000e+00, -1.000000e+00, -1.004697e-08,  1.000000e+00,
E           2.000000e+00])
E    y: array([-2, -1,  0,  1,  2])
        bounds     = Bounds(array([-5.12, -5.12, -5.12,  1.  ,  2.  ]), array([-2.  , -1.  ,  5.12,  5.12,  5.12]))
        bounds_old = [(-5.12, -2.0), (-5.12, -1.0), (-5.12, 5.12), (1.0, 5.12), (2.0, 5.12)]
        func       = <function TestDualAnnealing.test_bounds_class.<locals>.func at 0x7fffe25069e0>
        lw         = [-5.12, -5.12, -5.12, 1.0, 2.0]
        ret_bounds_class =      fun: 10.000000000000021
 message: ['Maximum number of iteration reached']
    nfev: 10199
    nhev: 0
     nit: 1... success: True
       x: array([-2.0000000e+00, -1.0000000e+00, -1.0046973e-08,  1.0000000e+00,
        2.0000000e+00])
        ret_bounds_list =      fun: 10.000000000000021
 message: ['Maximum number of iteration reached']
    nfev: 10199
    nhev: 0
     nit: 1... success: True
       x: array([-2.0000000e+00, -1.0000000e+00, -1.0046973e-08,  1.0000000e+00,
        2.0000000e+00])
        self       = <scipy.optimize.tests.test__dual_annealing.TestDualAnnealing object at 0x7fffe317d7b0>
        up         = [-2.0, -1.0, 5.12, 5.12, 5.12]
```


### SciPy/NumPy/Python version information

1.9.0  1.22.2  sys.version_info(major=3, minor=10, micro=2, releaselevel='final', serial=0)

--- a/scipy/optimize/tests/test__dual_annealing.py
+++ b/scipy/optimize/tests/test__dual_annealing.py
@@ -362,6 +362,6 @@ class TestDualAnnealing:
 
         # test that found minima, function evaluations and iterations match
         assert_allclose(ret_bounds_class.x, ret_bounds_list.x, atol=1e-8)
-        assert_allclose(ret_bounds_class.x, np.arange(-2, 3), atol=1e-8)
+        assert_allclose(ret_bounds_class.x, np.arange(-2, 3), atol=1e-7)
         assert_allclose(ret_bounds_list.fun, ret_bounds_class.fun, atol=1e-9)
         assert ret_bounds_list.nfev == ret_bounds_class.nfev

